src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -1042,14 +1043,23 @@ public void Joining(UniqueAddress node, ImmutableHashSet<string> roles)
             else
             {
                 var localMembers = _latestGossip.Members;
+                var localMember = localMembers.FirstOrDefault(m => m.Address == node.Address);
 
-                // check by address without uid to make sure that node with same host:port is not allowed
-                // to join until previous node with that host:port has been removed from the cluster
-                var alreadyMember = localMembers.Any(m => m.Address == node.Address);
-                var isUnreachable = !_latestGossip.Overview.Reachability.IsReachable(node);
-
-                if (alreadyMember) _log.Info("Existing member [{0}] is trying to join, ignoring", node);
-                else if (isUnreachable) _log.Info("Unreachable member [{0}] is trying to join, ignoring", node);
+                if (localMember != null && localMember.UniqueAddress == node)
+                {
+                    _log.Info("Existing member [{0}] is joining again.", node);
+                    if (!node.Equals(SelfUniqueAddress))
+                    {
+                        Sender.Tell(new InternalClusterAction.Welcome(SelfUniqueAddress, _latestGossip));
+                    }
+                }
+                else if (localMember != null)
+                {
+                    _log.Info("New incarnation of existing member [{0}] is trying to join. " +
+                        "Existing will be removed from the cluster and then new member will be allowed to join.", node);
+                    if (localMember.Status != MemberStatus.Down && localMember.Status != MemberStatus.Leaving && localMember.Status != MemberStatus.Exiting)
+                        Downing(localMember.Address);
```

Comment:
so is this where the magic happens for downing the unreachable node going forward? The "big fix" as it were?